### ISSUE TITLE
deadEndsToString

### ISSUE THREAD

**Issue by girishso** [brandly/elm-to-dot#4]
Run failed with `failed to parse: TODO deadEndsToString`

**Comment by brandly** [comment]
hi! does the `.elm` file have valid syntax? it compiles? if so, could you share the file (or the whole project) with me?

i'm spitting out an error here when `elm-syntax` fails to parse an Elm file. maybe [that repo](https://github.com/stil4m/elm-syntax) has a bug.

https://github.com/brandly/elm-to-dot/blob/e098a6a956464e26c985bf635101ac3054a539bd/src/Main.elm#L105-L106

---

however, the TODO comes from the [official `elm/parser`](https://github.com/elm/parser) via [`Parser.deadEndsToString`](https://github.com/elm/parser/blob/7506b07eaa93a93d13b508b948c016105b0953c8/src/Parser.elm#L157-L171). [the issue](https://github.com/elm/parser/issues/9) has been open since August 26, 2018. 

[pull](https://github.com/elm/parser/pull/16) [requests](https://github.com/elm/parser/pull/38) have been submitted. @evancz even [commented on one](https://github.com/elm/parser/pull/38#issuecomment-563344169), while not actually addressing the PR. it's all very confusing.

i didn't realize this when i used that function. like why is the function even there if it's useless?

tl;dr - maybe your `.elm` is invalid or maybe `elm-syntax` has a bug. regardless, `elm/parser` needs attention.

**Comment by brandly** [comment]
fixed in `v0.2.1`. you should get something like this now:

```shell
$ elm-to-dot src/Main.elm
failed to parse: ExpectingEnd at row:18 col:5

```

run `npm i -g elm-to-dot` again to update!

@girishso thanks for letting me know about this!

**Comment by brandly** [comment]
if you think it's failing on valid Elm code, let me know! it's almost certainly a problem with `elm-syntax`, and i could help track it down or file the issue on that repo. thanks again.
